common/src/test/java/com/datastrato/gravitino/json/TestDTOJsonSerDe.java

```diff
@@ -312,24 +314,66 @@ public void testPartitioningDTOSerDe() throws Exception {
-        ListPartitioningDTO.builder()
-            .withFieldNames(new String[][] {field1, field2})
-            // .withAssignment("p202304_California", p1Value)
-            // .withAssignment("p202304_Texas", p2Value)
+    LiteralDTO[][] pCaliforniaValue = {
+      {
+        LiteralDTO.builder().withValue("2023-04-01").withDataType(Types.DateType.get()).build(),
+        LiteralDTO.builder().withValue("San Francisco").withDataType(Types.StringType.get()).build()
+      },
+      {
+        LiteralDTO.builder().withValue("2023-04-01").withDataType(Types.DateType.get()).build(),
+        LiteralDTO.builder().withValue("San Diego").withDataType(Types.StringType.get()).build()
+      }
+    };
+    ListPartitionDTO pCalifornia =
+        ListPartitionDTO.builder()
+            .withName("p202304_California")
+            .withLists(pCaliforniaValue)
             .build();
 
+    LiteralDTO[][] pTexasValue = {
+      {
+        LiteralDTO.builder().withValue("2023-04-01").withDataType(Types.DateType.get()).build(),
+        LiteralDTO.builder().withValue("Houston").withDataType(Types.StringType.get()).build()
+      },
+      {
+        LiteralDTO.builder().withValue("2023-04-01").withDataType(Types.DateType.get()).build(),
+        LiteralDTO.builder().withValue("Dallas").withDataType(Types.StringType.get()).build()
+      }
+    };
+    ListPartitionDTO pTexas =
+        ListPartitionDTO.builder().withName("p202304_Texas").withLists(pTexasValue).build();
+
+    Partitioning listPart =
+        ListPartitioningDTO.of(
+            new String[][] {field1, field2}, new ListPartitionDTO[] {pCalifornia, pTexas});
+
     // construct range partition
-    // TODO: support assign partition value
-    Partitioning rangePart =
-        RangePartitioningDTO.builder()
-            .withFieldName(field1)
-            // .withRange("p20230101", "2023-01-01T00:00:00", "2023-01-02T00:00:00")
-            // .withRange("p20230102", "2023-01-01T00:00:00", null)
+    RangePartitionDTO p20230101 =
+        RangePartitionDTO.builder()
+            .withName("p20230101")
+            .withUpper(
+                LiteralDTO.builder()
+                    .withValue("2023-01-01")
+                    .withDataType(Types.DateType.get())
+                    .build())
+            .withLower(
+                LiteralDTO.builder()
+                    .withValue("2023-01-01")
+                    .withDataType(Types.DateType.get())
+                    .build())
             .build();
+    RangePartitionDTO p20230102 =
+        RangePartitionDTO.builder()
+            .withName("p20230102")
+            .withUpper(
+                LiteralDTO.builder()
+                    .withValue("2023-01-02")
+                    .withDataType(Types.DateType.get())
+                    .build())
+            .withLower(LiteralDTO.NULL)
+            .build();
+    Partitioning rangePart =
+        RangePartitioningDTO.of(field1, new RangePartitionDTO[] {p20230101, p20230102});
```

Comment:
Can you please also add some failure tests for your change in serde.

Reply:
Also, can you please add a mock test both in server and client side to see if the serde works well.

Reply:
added